lib/logstash/codecs/multiline.rb

```diff
@@ -126,7 +127,68 @@ class LogStash::Codecs::Multiline < LogStash::Codecs::Base
   # max_lines.
   config :max_bytes, :validate => :bytes, :default => "10 MiB"
 
+  # The accumulation of multiple lines will be converted to an event when either a
+  # matching new line is seen or there has been no new data appended for this time
+  # auto_flush_interval. No default.  If unset, no auto_flush
+  config :auto_flush_interval, :validate => :number
+
   public
+
+  class AutoFlushUnset
+    # NullObject pattern
+    def pending?
+      false
+    end
+
+    def stopped?
+      true
+    end
+
+    def method_missing(*args, &block)
+      self
+    end
+  end
```

Comment:
Have you considered defining all the possible methods instead of using `method_missing`?